x-pack/plugins/security_solution/public/detections/components/rules/threshold_input/index.tsx

```diff
@@ -53,7 +52,8 @@ const ThresholdInputComponent: React.FC<ThresholdInputProps> = ({
     () => ({
       fullWidth: true,
       noSuggestions: false,
-      options: getCategorizedFieldNames(browserFields),
+      options: browserFields.map((field) => ({ label: field.name })),
```

Comment:
Also, any perf considerations necessary here since we're dealing with what could be a quite large array? Doesn't look like `getCategorizedFieldNames` had cached anything either, so perhaps not that much of a hot-path

Reply:
just doing the map is O(n) vs `getCategorizedFieldNames` is O(nlgn). Ideally we wouldn't need to iterate over these at all here but the field component needs an array of `{label: string}`. I think we can take a look at this when we get performance tests set up for this.